daml-lf/scenario-interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/ScenarioRunner.scala

```diff
@@ -237,19 +237,19 @@ object ScenarioRunner {
 
   // The interface we need from a ledger during submission. We allow abstracting over this so we can play
   // tricks like caching all responses in some benchmarks.
-  trait LedgerApi[R] {
+  abstract class LedgerApi[R] {
```

Comment:
I think I asked you this before but I forgot: Why do you prefer `abstract class` over `trait`?

Reply:
@S11001001 told me that it is in general more efficient. So I started to use it blindly. 

After using this pattern blindly for some time, I like that because it is a simple pattern to avoid multi-intermittency : i.e. 
I tried to use only `abstract class` as super type,  and I  tried to use `trait` only for mixin (like for instance .

Reply:
> it is a simple pattern to avoid multi-intermittency

The best reason.

Reply:
seems reasonable, thanks for the explanation